pex/dependency_configuration.py

```diff
@@ -0,0 +1,166 @@
+            override for override in overrides if target.requirement_applies(override)
+        ]
+        if len(applicable_overrides) > 1:
+            raise ValueError(
+                "Invalid override configuration for target {target}.\n"
+                "More than one applicable override was found for {requirement}:\n"
+                "{overrides}".format(
+                    requirement=repr(str(requirement)),
+                    target=target.render_description(),
+                    overrides="\n".join(
+                        "{index}. {override}".format(index=index, override=repr(str(override)))
+                        for index, override in enumerate(applicable_overrides, start=1)
+                    ),
+                )
+            )
+        return applicable_overrides[0] if applicable_overrides else None
+
+    def merge(self, other):
+        # type: (DependencyConfiguration) -> DependencyConfiguration
+
+        excluded = OrderedSet(self.excluded)
+        excluded.update(other.excluded)
+
+        overridden = defaultdict(
+            OrderedSet
+        )  # type: DefaultDict[ProjectName, OrderedSet[Requirement]]
+        for project_name, overrides in itertools.chain(
+            self.overridden.items(), other.overridden.items()
+        ):
+            overridden[project_name].update(overrides)
+
+        return DependencyConfiguration(
+            excluded=tuple(excluded),
+            overridden={
+                project_name: tuple(overrides) for project_name, overrides in overridden.items()
+            },
+        )
+
+
+def register(parser):
+    # type: (_ActionsContainer) -> None
+    """Register dependency configuration options with the given parser.
+
+    :param parser: The parser to register dependency configuration options with.
+    """
+    parser.add_argument(
+        "--exclude",
+        dest="excluded",
+        default=[],
+        type=str,
+        action="append",
+        help=(
+            "Specifies a requirement to exclude from the built PEX. Any distribution included in "
+            "the PEX's resolve that matches the requirement is excluded from the built PEX along "
+            "with all of its transitive dependencies that are not also required by other "
+            "non-excluded distributions. At runtime, the PEX will boot without checking the "
+            "excluded dependencies are available (say, via `--inherit-path`). This option can be "
+            "used multiple times."
+        ),
+    )
+    parser.add_argument(
+        "--override",
+        dest="overridden",
+        default=[],
+        type=str,
+        action="append",
+        help=(
+            "Specifies a transitive requirement to override when resolving. Any distribution "
+            "requirement in the PEX's resolve that matches the override project name is replaced "
+            "with the given override requirement. This option can be used multiple times."
+        ),
+    )
```

Comment:
@benjyw I wanted to draw your attention to this change retroactively. The `--exclude` and `--override` options are very different from `--constraints`. With `--constraint` you can only narrow the requirements of some transitive dependency. With `--exclude` and `--override` you can actually _edit_ the requirements of a transitive dependency.